signing/src/main/java/tech/pegasys/web3signer/signing/BlsBKeystoreBulkLoader.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing;
+
+import tech.pegasys.signers.bls.keystore.KeyStore;
+import tech.pegasys.signers.bls.keystore.KeyStoreLoader;
+import tech.pegasys.signers.bls.keystore.KeyStoreValidationException;
+import tech.pegasys.signers.bls.keystore.model.KeyStoreData;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.bls.BLSSecretKey;
+import tech.pegasys.web3signer.signing.config.metadata.SignerOrigin;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class BlsBKeystoreBulkLoader {
+  private static final Logger LOG = LogManager.getLogger();
+
+  public Collection<ArtifactSigner> loadKeystoresUsingPasswordDir(
+      final Path keystoresDirectory, final Path passwordsDirectory) {
+    final List<Path> keystoreFiles = keystoreFiles(keystoresDirectory);
+    return keystoreFiles.parallelStream()
+        .map(
+            keystoreFile ->
+                createSignerForKeystore(
+                    keystoreFile,
+                    key -> Files.readString(passwordsDirectory.resolve(key + ".txt"))))
+        .flatMap(Optional::stream)
+        .collect(Collectors.toList());
+  }
+
+  public Collection<ArtifactSigner> loadKeystoresUsingPasswordFile(
+      final Path keystoresDirectory, final Path passwordFile) {
+    final List<Path> keystoreFiles = keystoreFiles(keystoresDirectory);
+    final String password;
+    try {
+      password = Files.readString(passwordFile);
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to read the password file", e);
```

Comment:
nit: You can instead wrap in `UncheckedIOException` instead of `IllegalStateException`.